lib/Arch/Sleigh/X86Arch.cpp

```diff
@@ -32,22 +32,49 @@
 namespace remill {
 namespace sleigh::x86 {
 
-class SleighX86Arch final : public SleighArch, public remill::X86ArchBase {
+class SleighX86Decoder final : public SleighDecoder {
+ public:
+  SleighX86Decoder() = delete;
+  SleighX86Decoder(const remill::Arch &arch)
+      : SleighDecoder(
+            arch, kArchX86_SLEIGH == arch.arch_name ? "x86.sla" : "x86-64.sla",
```

Comment:
static methods (maybe in parent?)/standalone functions for those ternaries? Sound useful in some other places.